### ISSUE TITLE
magit-todos--scan-with-rg doesn't generate TODO sections

### ISSUE THREAD

**Issue by ST-Saint** [alphapapa/magit-todos#141]
I tried these magit-todo-scanners
```
magit-todos--scan-with-find|grep
magit-todos--scan-with-git-diff
magit-todos--scan-with-git-grep
magit-todos--scan-with-rg
```

Only `magit-todos--scan-with-git-grep` generates the TODO items. 

However, I also check the rg commands and it seems reasonable
```lisp
:command ("nice" "-n5" "rg" "--no-heading" "--line-number" "--glob" "!.git/" "^(\\*+)[[:blank:]]+(DEBUG|FIXME|HOLD|STUB|TODO)[[:space:]]+(.+)|(?:^|[[:blank:]]+)(DEBUG|FIXME|HOLD|STUB|TODO)(?:\\([^)]+\\))?:(?:[[:blank:]]+(.+))?" "./")
```

and when I executed it manually, I also get the expected output
```shell
sh -c 'nice -n5 rg --no-heading --line-number --glob "!.git/" "^(\\*+)[[:blank:]]+(DEBUG|FIXME|HOLD|STUB|TODO)[[:space:]]+(.+)|(?:^|[[:blank:]]+)(DEBUG|FIXME|HOLD|STUB|TODO)(?:\\([^)]+\\))?:(?:[[:blank:]]+(.+))?" "./"
./config.org:512:TODO(magit todo): test
```

Not sure how to further debug.

Tested on emacs 28.2 and 30.0

**Comment by alphapapa** [comment]
Hello,

If you browse some of the old issues on this repo's tracker, you can find similar discussions.  Often the problem turns out to be a misunderstanding about how to-do items are supposed to be formatted in order to be recognized; or the user may not have configured the options to match the formatting of items in a certain source code repository.  And occasionally the problem may be related to software versions or the OS/platform, which you haven't mentioned here.

Generally I would suggest trying to reproduce the problem on other source code repos, e.g. you could clone this package's repo and see if items are displayed from its files.  If that works, then you'll know that the problem is a matter of configuration or item formatting.

**Comment by ST-Saint** [comment]
Thanks for your reply! 

The test `TODO(magit todo)` is actually from a code block. I also tried different TODO formats as well as `magit-todos` repo, and the TODO did not show. I assume it is a software or system issue.

However, just after a trivial reboot, it somehow start to work today. I did not change the configuration and I have no idea what is going on but I cannot reproduce the bug again. Anyways, thanks for your help.
